src/Controller/Rest/ApiController.php

```diff
@@ -44,39 +46,76 @@ public function getFreeTimesByDate(string $date)
     {
 
         //Reikia gauti laisvas vietas pagal data ir transforminti i apacioje esanti masyva
-        $dateArr = [
-            '0100-0140' =>  4,
-            '0500-0600' =>  2,
-            '0600-0650' =>  1,
-        ];
+//        $dateArr = [
+//            '0100-0140' =>  1,
+//            '0500-0600' =>  1,
+//            '0600-0650' =>  1,
+//        ];
+
+        $sessionRepo = $this->getDoctrine()->getRepository(Session::class);
+
+        $freeSlotOfDayColl = $sessionRepo->findFreeTimeSlotByDate($date);
 
-        return View::create($date, Response::HTTP_OK);
+        $freeSlotOfDayArray = [];
+
+        //transform collection data to proper format array
+        foreach ($freeSlotOfDayColl as $value){
+            $freeSlotOfDayArray[str_replace(':', "",$value->getStartsAt()->format('H:i').'-'.$value->getEndsAt()->format('H:i'))] = 1;
+        }
+
+        return View::create($freeSlotOfDayArray, Response::HTTP_OK);
 
     }
 
     /**
      * Create an booking
      * @Rest\Post("/calendar")
      * @param Request $request
+     * @return View
      */
     public function postBookingData(Request $request){
 
+        //Find slot for booking
+        $sessionRepo = $this->getDoctrine()->getRepository(Session::class);
+
+        $times = explode("-", $request->get('timeslot'));
+
+        $startTime = strpos($times[0], ':', 2).':00';
+        $endTime =   strpos($times[1], ':', 2).':00';
+
+        $slot = $sessionRepo->checkSlotFree($request->get('date'), $startTime, $endTime);
```

Comment:
metodai, kurie kazka tikrina vadinami `is....` siuo atveju `isSlotFree`, bet tai nera repositorijos metodas - repositorija neturi tureti jokios logikos pas save tai gali buti kazkokio tai session managerio metodas (turetu bent jau), kuri jus cia kviesti turetumet, na arba bent jau atskiras sito controllerio metodas, kuris pas save jau kviecia repositorijos metoda, kuris surenka slotus reikiamus ir jau pas save juos apdoroja. Dar vienas dalykas, metodas su tokiu namingu turi grazinti bool, o pas jus is kazkur cia slotas atsiranda. Cia kaip getteris / finderis / creat'eris atrodo labiau...